irispy/iris_tools.py

```diff
@@ -165,53 +166,247 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
+        iris_response["AREA_SJI"] = np.zeros(iris_response["AREA_SJI"].shape)
+
+        # 1. FUV SG effective areas
+        lambran_fuv = np.array([[133.1, 135.9],[138.8, 140.8]])
+        # Rough SG spectral ranges.  Setting effective area to 0 outside of these.
+        shp_fuv = iris_response["COEFFS_FUV"].shape
+        # Time-dependent response for shp_0[0] = 3 wavelengths
+        iris_fit_fuv = np.zeros((n_time_obs, shp_fuv[0]))
+        detector_fuv = "FUV"
+        for j in range(shp_fuv[0]):
+            iris_fit_fuv[:, j] = fit_iris_xput(time_obs, iris_response["C_F_TIME"], iris_response["COEFFS_FUV"][j, :, :])
+        # Interpolate onto lambda grid, separately for each of the two FUV CCD's.
+        for j in range(2):
+            w_fuv = np.logical_and(iris_response["LAMBDA"]/u.nm >= lambran_fuv[j, 0], iris_response["LAMBDA"]/u.nm  <= lambran_fuv[j, 1])
+            for k in range(n_time_obs):
+                interpol_fuv = scipy.interpolate.interp1d(iris_response["C_F_LAMBDA"][j:j+2], np.squeeze(iris_fit_fuv[k, j:j+2]), fill_value='extrapolate')
+                iris_response["AREA_SG"][0, w_fuv] = interpol_fuv(iris_response["LAMBDA"][w_fuv])
+    
+        # 2. NUV SG effective areas
+        lambran_nuv = np.array([278.2, 283.5])
+        # Rough SG spectral ranges.  Setting effective area to 0 outside of these.
+        shp_nuv = iris_response["COEFFS_NUV"].shape
+        # Time-dependent response for shp_1[0] wavelengths
+        iris_fit_nuv = np.zeros((n_time_obs, shp_nuv[0]))
+        detector_nuv = "NUV"
+        for j in range(shp_nuv[0]):
+            iris_fit_nuv[:, j] = fit_iris_xput(time_obs, iris_response["C_N_TIME"], iris_response["COEFFS_NUV"][j, :, :])
+            # Interpolate onto lambda grid
+            w_nuv = np.where(np.logical_and(iris_response["LAMBDA"]/u.nm >= lambran_nuv[0], iris_response["LAMBDA"]/u.nm <= lambran_nuv[1]))
+        for k in range(n_time_obs):
+            interpol_nuv =  scipy.interpolate.interp1d(iris_response["C_N_LAMBDA"][:], np.squeeze(iris_fit_nuv[k, :]), fill_value='extrapolate')
+            iris_response["AREA_SG"][1, w_nuv] = interpol_nuv(iris_response["LAMBDA"][w_nuv])
+        
+        # 3. SJI effective areas
+        shp_sji = iris_response["COEFFS_SJI"].shape
+        area_pre_launch = iris_response["GEOM_AREA"]
+        for j in range(shp_sji[0]):
+            # Calculate pre-launch area from the individual elements
+            area_pre_launch = iris_response["GEOM_AREA"] * np.prod(iris_response["ELEMENTS"]["trans"][iris_response["INDEX_EL_SJI"]], axis=1)
+            # Time dependent response
+            iris_fit_sji = fit_iris_xput(time_obs, iris_response["C_S_TIME"][j, :, :], iris_response["COEFFS_SJI"][j, :, :])
+            # Time dependent profiles
+            for k in range(n_time_obs):
+                iris_response["AREA_SJI"][j, :] = area_pre_launch[j] * iris_fit_sji[k]
+            area_sji = iris_response["GEOM_AREA"]
+            for nuv in range(2):
```

Comment:
now this is where the condition starts for versions greater than 3 so need
`elif int(iris_response['VERSION']) >3:`